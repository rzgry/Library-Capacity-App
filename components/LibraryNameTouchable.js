import React from 'react';
import Touchable from 'react-native-platform-touchable';
import { StyleSheet, Text, View } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  optionsTitleText: {
    fontSize: 16,
    marginLeft: 15,
    marginTop: 9,
    marginBottom: 12,
  },
  optionIconContainer: {
    marginRight: 9,
  },
  option: {
    backgroundColor: '#fdfdfd',
    paddingHorizontal: 15,
    paddingVertical: 15,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: '#EDEDED',
  },
  optionText: {
    fontSize: 15,
    marginTop: 1,
  },
});

const LibraryNameTouchable = ({ libraryName, onPress }) => (
  <Touchable style={styles.option} background={Touchable.Ripple('#ccc', false)} onPress={onPress}>
    <View style={{ flexDirection: 'row' }}>
      <View style={styles.optionTextContainer}>
        <Text style={styles.optionText}>{libraryName}</Text>
      </View>
    </View>
  </Touchable>
);

LibraryNameTouchable.propTypes = {
  libraryName: PropTypes.string.isRequired,
  onPress: PropTypes.func,
};

LibraryNameTouchable.defaultProps = {
  onPress: () => {},
};

export default LibraryNameTouchable;
