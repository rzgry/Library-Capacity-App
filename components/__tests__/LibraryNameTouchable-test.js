import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import LibraryNameTouchable from '../LibraryNameTouchable';

it('renders correctly', () => {
  const tree = renderer.create(<LibraryNameTouchable libraryName="Test library" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('handles being pressed correctly', () => {
  let pressed = false;

  const wrapper = shallow(
    <LibraryNameTouchable
      libraryName="Test library"
      onPress={() => {
        pressed = true;
      }}
    />,
  );

  wrapper.props().onPress();

  expect(pressed).toBe(true);
});
