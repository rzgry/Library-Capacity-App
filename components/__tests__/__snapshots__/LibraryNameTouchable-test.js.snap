// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fdfdfd",
      "borderBottomColor": "#EDEDED",
      "borderBottomWidth": 0.5,
      "opacity": 1,
      "paddingHorizontal": 15,
      "paddingVertical": 15,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 15,
            "marginTop": 1,
          }
        }
      >
        Test library
      </Text>
    </View>
  </View>
</View>
`;
